internal/bundler/apko.go

```diff
@@ -0,0 +1,153 @@
+package bundler
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+
+	apko_build "chainguard.dev/apko/pkg/build"
+	apko_oci "chainguard.dev/apko/pkg/build/oci"
+	apko_types "chainguard.dev/apko/pkg/build/types"
+	"chainguard.dev/apko/pkg/tarfs"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/empty"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+)
+
+// apko is a bundler that uses the local machine to build the
+// image using apko.
+type apko struct {
+	arch       apko_types.Architecture
+	apkoConfig apko_types.ImageConfiguration
+	ropts      []remote.Option
+}
+
+type ApkoOpt func(*apko) error
+
+func NewApko(opts ...ApkoOpt) (Bundler, error) {
+	a := &apko{
+		arch: apko_types.ParseArchitecture(runtime.GOARCH),
```

Comment:
For remote harnesses, could the arch of the provider and the container runtime differ?

Reply:
yeah good point.

I added another optional to override the opt, but don't call it anywhere yet. I think in the future we'll expose it as a harness scoped attribute that can be overridden by the harnesses that will need it.